### PR TITLE
Consumer and service for update status sms

### PR DESCRIPTION
**Pull request by tba76**
## Related Issue(s)
- #360 

## Verification
- [ ] **Your** code builds clean without any errors or warnings
- [ ] Manual testing done (required)
- [ ] Relevant automated test added (if you find this hard, leave it and we'll help out)
- [ ] All tests run green

## Documentation
- [ ] User documentation is updated with a separate linked PR in [altinn-studio-docs.](https://github.com/Altinn/altinn-studio-docs) (if applicable)

### CODE REVIEW

File: src/Altinn.Notifications.Core/Models/Notification/EmailSendOperationResult.cs [hunk 1]
```diff
@@ -5,9 +5,9 @@
 namespace Altinn.Notifications.Core.Models.Notification;
 
 /// <summary>
-/// A class representing a send operation update object
+/// A class representing a email send operation update object
```
Comment (acn-sbuad) [inline src/Altinn.Notifications.Core/Models/Notification/EmailSendOperationResult.cs:8]:
```suggestion
/// A class representing an email send operation result object
```

File: src/Altinn.Notifications.Core/Models/Notification/SmsSendOperationResult.cs [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+    /// <summary>
+    /// The reference to the delivery in sms gateway
+    /// </summary>
+    public string GatewayReference { get; set; } = string.Empty;
+
+    /// <summary>
+    /// The sms send result
+    /// </summary>
+    public SmsNotificationResultType? SendResult { get; set; }
```
Comment (acn-sbuad) [inline src/Altinn.Notifications.Core/Models/Notification/SmsSendOperationResult.cs:25]: does it make sense that this is nullable? We will never update just the gateway reference as we only get a gateway reference if the sending has been accepted by LinkMobility.

Comment (tba76) [reply src/Altinn.Notifications.Core/Models/Notification/SmsSendOperationResult.cs:25]: No :)

File: test/Altinn.Notifications.IntegrationTests/Notifications/EmailNotificationsController/GetTests.cs [hunk 3]
```diff
@@ -24,29 +24,27 @@ public class GetTests : IClassFixture<IntegrationTestWebApplicationFactory<Contr
...
-            _notificationId = persistedNotification.Id;
+            await Task.CompletedTask;
         }
 
         async Task IAsyncLifetime.DisposeAsync()
         {
-            string sql = $"delete from notifications.orders where alternateid= '{_orderId}'";
-
-            await PostgreUtil.RunSql(sql);
+            if (orderIdsToDelete.Count == 0)
```
Comment (acn-sbuad) [inline test/Altinn.Notifications.IntegrationTests/Notifications/EmailNotificationsController/GetTests.cs:43]:
```suggestion
            if (orderIdsToDelete.Count != 0)
```

**Review by acn-sbuad — APPROVED** [review]
🎈

**Pull request merged**